core/dbt/graph/selector.py

```diff
@@ -177,10 +180,14 @@ def _is_graph_member(self, unique_id: UniqueId) -> bool:
 
         node = self.manifest.nodes[unique_id]
 
-        if self.include_empty_nodes:
-            return node.config.enabled
+        return node.config.enabled
+
+    def _is_empty_node(self, unique_id: UniqueId) -> bool:
+        if unique_id in self.manifest.nodes:
```

Comment:
Why do we need this check here?

Reply:
Fair point. I was writing defensive code (similar to `_is_match` and `_is_graph_member`), but this method should only ever be called on a `unique_id` returned (within a set) from resolving node selection criteria against the current manifest.